agent/command/s3_copy.go

```diff
@@ -188,43 +243,76 @@ func (c *s3copy) s3Copy(ctx context.Context,
 			s3CopyFile.Source.Path, s3CopyFile.Destination.Bucket,
 			s3CopyFile.Destination.Path)
 
-		s3CopyReq := apimodels.S3CopyRequest{
-			AwsKey:              c.AwsKey,
-			AwsSecret:           c.AwsSecret,
-			S3SourceRegion:      s3CopyFile.Source.Region,
-			S3SourceBucket:      s3CopyFile.Source.Bucket,
-			S3SourcePath:        s3CopyFile.Source.Path,
-			S3DestinationRegion: s3CopyFile.Destination.Region,
-			S3DestinationBucket: s3CopyFile.Destination.Bucket,
-			S3DestinationPath:   s3CopyFile.Destination.Path,
-			S3DisplayName:       s3CopyFile.DisplayName,
-			S3Permissions:       s3CopyFile.Permissions,
-		}
-
-		responseString, err := comm.S3Copy(ctx, td, &s3CopyReq)
+		s3CopyReq.AwsKey = c.AwsKey
+		s3CopyReq.AwsSecret = c.AwsSecret
 
-		if responseString != "" {
-			logger.Task().Infof("s3Copy response: %s", responseString)
+		// Now copy the file into the permanent location
+		srcOpts := pail.S3Options{
+			Credentials: pail.CreateAWSCredentials(s3CopyReq.AwsKey, s3CopyReq.AwsSecret, ""),
+			Region:      s3CopyReq.S3SourceRegion,
+			Name:        s3CopyReq.S3SourceBucket,
+			Permissions: pail.S3Permissions(s3CopyReq.S3Permissions),
 		}
 
+		srcBucket, err := pail.NewS3MultiPartBucketWithHTTPClient(client, srcOpts)
 		if err != nil {
-			err = errors.Wrap(err, "s3 push copy failed")
-			logger.Execution().Error(err)
-
-			if s3CopyFile.Optional {
-				logger.Execution().Errorf("file '%s' is optional, continuing",
-					s3CopyFile.DisplayName)
-				continue
-			} else {
-				return errors.Wrapf(err, "failed to push %s to %s",
-					s3CopyFile.Source.Path, s3CopyFile.Destination.Bucket)
-			}
-
+			connectionErr := errors.Wrap(err, "S3 copy failed, could not establish connection to source bucket")
+			grip.Error(connectionErr)
```

Comment:
can we remove this grip log (and at 277)? These aren't very useful since they're just logged to the agent